### PR TITLE
DEPR: Enforce default of numeric_only=False in DataFrame methods

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -10090,15 +10081,11 @@ def cov(
 
             .. versionadded:: 1.1.0
 
-        numeric_only : bool, default True
+        numeric_only : bool, default False
             Include only `float`, `int` or `boolean` data.
 
             .. versionadded:: 1.5.0
 
-            .. deprecated:: 1.5.0
```
Comment (jbrockmendel) [inline pandas/core/frame.py:10098]: versionchanged?

File: doc/source/whatsnew/v2.0.0.rst [hunk 2]
```diff
@@ -502,7 +502,7 @@ Removal of prior version deprecations/changes
 - Removed ``na_sentinel`` argument from :func:`factorize`, :meth:`.Index.factorize`, and :meth:`.ExtensionArray.factorize` (:issue:`47157`)
 - Changed behavior of :meth:`DataFrameGroupBy.apply` and :meth:`SeriesGroupBy.apply` so that ``group_keys`` is respected even if a transformer is detected (:issue:`34998`)
 - Enforced deprecation ``numeric_only=None`` (the default) in DataFrame reductions that would silently drop columns that raised; ``numeric_only`` now defaults to ``False`` (:issue:`41480`)
--
+- Changed default of ``numeric_only`` to ``False`` in all DataFrame methods with that argument (:issue:`46096`, :issue:`46906`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v2.0.0.rst:505]: Any Series methods affected?

**Comment by rhshadrach** [comment]
Thanks @jbrockmendel - changed the removal of deprecated to versionchanged and added a line about Series.rank in the whatsnew - that's the only Series method that didn't default to numeric_only=False.

**Review by mroeschke — APPROVED** [review]
LGTM @jbrockmendel merge when ready

**Review by jbrockmendel — APPROVED** [review]

**Comment by jbrockmendel** [comment]
thanks @rhshadrach

**Pull request merged**